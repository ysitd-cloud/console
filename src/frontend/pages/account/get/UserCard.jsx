import React from 'react';
import PropTypes from 'prop-types';
import Card from '../../../components/react/Card';
import Flex from '../../../components/react/Flex';
import Row from '../../../components/react/Row';

export default function UserCard({ avatarUrl, displayName, username }) {
  return (
    <Card>
      <Row>
        <Flex className="xs12 md2 lg1">
          <img src={avatarUrl} alt="Avatar Icon" style={{ minHeight: '50px' }} />
        </Flex>
        <Flex className="xs12 md8 lg11">
          <div className="card__title pb-0">
            <div className="headline mb-0">
              <span className="bold">Name: </span>{ displayName }
            </div>
          </div>
          <div className="card__text pt-1">
            <div><span className="bold">Username: </span>{ username }</div>
          </div>
        </Flex>
      </Row>
    </Card>
  );
}

UserCard.propTypes = {
  avatarUrl: PropTypes.string.isRequired,
  displayName: PropTypes.string.isRequired,
  username: PropTypes.string.isRequired,
};
