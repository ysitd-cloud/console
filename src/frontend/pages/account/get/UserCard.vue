<template>
    <v-card>
        <v-layout row>
            <v-flex xs12 md2 lg1>
                <img :src="avatarUrl" />
            </v-flex>
            <v-flex xs12 md8 lg11>
                <v-card-title class="pb-0">
                    <div class="headline mb-0">
                        <span class="bold">Name: </span>{{ name }}</div>
                </v-card-title>
                <v-card-text class="pt-1">
                    <div>
                        <span class="bold">Username: </span>{{ username }}</div>
                </v-card-text>
            </v-flex>
        </v-layout>
    </v-card>
</template>

<style scoped lang="stylus">
    img {
        min-height: 50px
    }
    .bold {
        font-weight: 500
    }
    .headline {
        font-weight: normal
    }
</style>

<script>
  export default {
    props: {
      name: {
        type: String,
        required: true,
      },
      avatarUrl: {
        type: String,
        required: true,
      },
      username: {
        type: String,
        required: true,
      },
    },
  };
</script>
