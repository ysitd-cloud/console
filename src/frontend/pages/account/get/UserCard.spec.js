import React from 'react';
import renderer from 'react-test-renderer';
import UserCard from './UserCard';

/* globals describe, expect, it */

describe('React Card Component', () => {
  it('match snapshot', () => {
    const component = renderer.create(<UserCard avatarUrl="foo.jpg" displayName="bar" username="foobar" />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
