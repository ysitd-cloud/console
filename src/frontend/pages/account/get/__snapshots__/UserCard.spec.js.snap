// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`React Card Component match snapshot 1`] = `
<div
  className="card"
  style={
    Object {
      "height": "auto",
    }
  }
>
  <div
    className="layout row"
  >
    <div
      className="flex xs12 md2 lg1"
      style={Object {}}
    >
      <img
        alt="Avatar Icon"
        src="foo.jpg"
        style={
          Object {
            "minHeight": "50px",
          }
        }
      />
    </div>
    <div
      className="flex xs12 md8 lg11"
      style={Object {}}
    >
      <div
        className="pb-0 card__title"
      >
        <div
          className="headline mb-0"
        >
          <span
            className="bold"
          >
            Name: 
          </span>
          bar
        </div>
      </div>
      <div
        className="card__text pt-1"
      >
        <div>
          <span
            className="bold"
          >
            Username: 
          </span>
          foobar
        </div>
      </div>
    </div>
  </div>
</div>
`;
